### PR TITLE
AUTH-1253: log aws request id and psid to logging context

### PR DESCRIPTION
**Pull request by dbes-gds**
## What?

Log aws request id and psid to logging context.

## Why?

Use the aws request id to exclude smoke tester traffic from queries and reports.

### CODE REVIEW

**Review by jamietanna — COMMENTED** [review]
Is this something we want to add tests for?

Also, is there any reason we're using this means over `MDC`?

**Comment by dbes-gds** [comment]
@jamietanna will add some tests.  MDC is just a wrapper around ThreadContext so I don't think makes much difference.

**Comment by mrwilson** [comment]
@jamietanna MDC is an slf4j thing, as I understand, which we aren't using. ThreadContext is the equivalent log4j2 concept.

> Log4j 2 continues with the idea of the MDC and the NDC but merges them into a single Thread Context. The Thread Context Map is the equivalent of the MDC and the Thread Context Stack is the equivalent of the NDC.

https://logging.apache.org/log4j/2.x/manual/thread-context.html

@dbes-gds The `CaptureLoggingExtension` should have all the magic needed to do assertions against logging.

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java [hunk 1]
```diff
@@ -112,6 +115,8 @@ public APIGatewayProxyResponseEvent handleRequest(
                                     ipAddress,
                                     AuditService.UNKNOWN,
                                     persistentSessionId);
+                            attachLogFieldToLogs(PERSISTENT_SESSION_ID, persistentSessionId);
```
Comment (mrwilson) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java:118]: We need to add `.flatMap(InputSanitiser::sanitiseBase64)` to the method `getExistingOrCreateNewPersistentSessionId`. We did this in `PersistentIdHelper`.

Maybe there's some refactoring to do here.

**Review by JHjava — APPROVED** [review]

**Pull request merged**